spec/classes/config_spec.rb

```diff
@@ -691,5 +691,11 @@
 
       it { expect { is_expected.to contain_class('nginx::config') }.to raise_error(Puppet::Error, %r{\$nginx_error_log_severity must be debug, info, notice, warn, error, crit, alert or emerg}) }
     end
+
+    context 'when log_dir is non-default' do
+      let(:params) { { log_dir: '/foo/bar' } }
+
+      it { is_expected.to contain_file('/foo/bar').with(ensure: 'directory') }
```

Comment:
could you add tests for the $http_access_log and $nginx_error_log variables as well? not sure if they contain the correct path.